Yoast/wordpress-seo
# P1-1277 - Implement tooling standards in js package

## Context


* Implement our tooling standards in the `js` package
  * `@yoast/babel-preset`, `eslint-config-yoast`, `@yoast/jest-preset` and `@yoast/browserslist-config`

## Summary


This PR can be summarized in the following changelog entry:

* Implements our new tooling standards in the 'js' package

## Relevant technical choices:

* ~Use the transpiled code only on production build~
  * ~Running the build command from the 'js' package on every file change when using i.e the watch command would negatively impact the development speed drastically.~
* Removed `jsSrc` and `cssDist` from `config/webpack/paths.js` because they were unused.
* Decided to revert the Babel config after talking with Nolle. Seems simpler to just keep using Webpack.
  * Downside is that this picks up the Babel config in the root. However, the test still use the Babel config in the JS folder. Something to talk about at a later stage.
* Use versions instead of file in our root
  * File copies over at the time of install. Using a version makes it a symlink because of the workspace. That seems preferable, and a must if we work on the packages.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Testing Eslint
  * Navigate to `packages/js` and run `yarn lint`
  * Verify there are now `141 warnings`
* Testing Jest
  * Navigate to `packages/js` and run `yarn test`
  * Verify `Test Suites: 55 passed` without any errors

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Check if `grunt build`, `grunt artifact` and `yarn lint` still work as expected.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes P1-1277
